docs/proposals/multi-platform.md

```diff
@@ -0,0 +1,554 @@
+
+Also, by including other platforms to the Kubernetes party a healthy competition between platforms can/will take place.
+
+Every platform obviously has both pros and cons. By adding the option to make clusters of mixed platforms, the end user may take advantage of the good sides of every platform.
+
+## Use Cases
+
+For a large enterprise where compute power is the king, one may imagine the following combination:
+ - `linux/amd64`: For running most of the general-purpose computing tasks, cluster addons, etc.
+ - `linux/ppc64le`: For running highly-optimized software; especially massive compute tasks
+ - `windows/amd64`: For running services that are only compatible on windows; e.g. business applications written in C# .NET
+
+For a mid-sized business where efficency is the kind, this could be a combination:
+ - `linux/amd64`: For running most of the general-purpose computing tasks, plus tasks that require very high single-core performance.
+ - `linux/arm64`: For running webservices and high-density tasks => the cluster could autoscale in a way that `linux/amd64` machines could hibernate at night in order to minimize power usage.
+
+For a small business or university, arm could be sufficent:
+ - `linux/arm`: Draws very little power, and could run web sites and app backends efficently on Scaleway for example.
+
+And last but not least; Raspberry Pi's should be used for [education at universities](http://kubecloud.io/) and is great for demoing Kubernetes' features at conferences.
+
+## Main proposal
+
+### Release binaries for all platforms
+
+First and foremost, binaries have to be released for all platforms.
+This affects build-release tools; but is quite straightforward to implement.
+
+Since Kubernetes release and build jobs run on `linux/amd64`, binaries have to be cross-compiled and Docker images should be cross-built.
+Builds should be run in a Docker container in order to get reproducible builds; and `gcc` should be installed for all platforms inside that image (called `kube-cross` for reference)
+
+All released binaries should be uploaded to `https://storage.googleapis.com/kubernetes-release/release/${version}/bin/${os}/${arch}/${binary}`
+
+That is a fairly long topic. If you're interested how to do cross-compile, see [details about cross-compilation](#cross-compilation-details)
+
+### Support all platforms in a "run everywhere" deployment
+
+The easiest way of running Kubernetes on another architecture is probably by using the docker or the docker-multinode deployment. Of course, you may choose whatever deployment you want, the binaries are easily downloadable from the URL above. But [docker-multinode](https://github.com/kubernetes/kube-deploy/tree/master/docker-multinode) should be a "just works" multi-platform solution with docker as the only real dependency.
+Note: That doesn't mean _everything_ "just works" with the docker-multinode setup, but basic functionality is there.
```

Comment:
I'm not confident that docker multinode is an acceptable production solution. It might be worth mentioning that. It is mainly meant for "kick the tires" experience.